drift/test/manager/manager_selectable_test.dart

```diff
@@ -0,0 +1,217 @@
+          store: Value(stores[1].id),
+          price: Value(15.0))),
+
+      // Target - Grocery
+
+      // More expensive groceries
+      await db.managers.listing.create((o) => o(
+          product: Value(products[3].id),
+          store: Value(stores[1].id),
+          price: Value(10.0))),
+      await db.managers.listing.create((o) => o(
+          product: Value(products[4].id),
+          store: Value(stores[1].id),
+          price: Value(20.0))),
+
+      // Target - Clothing
+
+      // Does not have any shirts or pants
+      // await db.managers.listing.create((o) => o(
+      //     product: Value(products[5].id),
+      //     store: Value(stores[1].id),
+      //     price: Value(20.0))),
+      // await db.managers.listing.create((o) => o(
+      //     product: Value(products[6].id),
+      //     store: Value(stores[1].id),
+      //     price: Value(30.0))),
+      await db.managers.listing.create((o) => o(
+          product: Value(products[7].id),
+          store: Value(stores[1].id),
+          price: Value(5.0))),
+      await db.managers.listing.create((o) => o(
+          product: Value(products[8].id),
+          store: Value(stores[1].id),
+          price: Value(10.0))),
+
+      // Costco - Electronics
+      // Much cheaper electronics
+      await db.managers.listing.create((o) => o(
+          product: Value(products[0].id),
+          store: Value(stores[2].id),
+          price: Value(50.0))),
+      await db.managers.listing.create((o) => o(
+          product: Value(products[1].id),
+          store: Value(stores[2].id),
+          price: Value(100.0))),
+      await db.managers.listing.create((o) => o(
+          product: Value(products[2].id),
+          store: Value(stores[2].id),
+          price: Value(2.50))),
+
+      // Costco - Grocery
+
+      // More expensive groceries
+      await db.managers.listing.create((o) => o(
+          product: Value(products[3].id),
+          store: Value(stores[2].id),
+          price: Value(20.0))),
+      await db.managers.listing.create((o) => o(
+          product: Value(products[4].id),
+          store: Value(stores[2].id),
+          price: Value(900.0)))
+    ];
+
+    final getAllStores = db.managers.store;
+    final getAllStoresWithFilter =
+        db.managers.store.filter((f) => f.id.not(10));
+    final getAllStoresWithOrdering =
+        db.managers.store.orderBy((f) => f.id.asc());
+    final getAllStoresWithFilterAndOrdering = db.managers.store
+        .filter((f) => f.id.not(10))
+        .orderBy((f) => f.id.asc());
+    final getAllStoresWithFilterAndOrderingWithReferences = db.managers.store
+        .filter((f) => f.id.not(10))
+        .orderBy((f) => f.id.asc())
+        .withReferences();
+
+    void testManager<T, M>(
+        BaseTableManager<dynamic, dynamic, T, dynamic, dynamic, dynamic,
+                dynamic, M, T>
+            selectable) {
+      expect(selectable.get().then((v) => v.length), completion(3));
```

Comment:
I think it would be easier to make `testManager` async and then do `expect((await selectable.get()).length, 3)`. Or, even better `expect(await selectable.get(), hasLengh(3))`